src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherDocumentServiceImpl.java

```diff
@@ -297,9 +303,31 @@ private RecurringDisbursementVoucherPDPStatus buildRecurringDisbursementVoucherP
         pdpStatus.setPdpStatus(disbursementVoucherDocument.getDisbursementVoucherPdpStatus());
         pdpStatus.setPaymentDetailDocumentType(disbursementVoucherDocument.getPaymentDetailDocumentType());
         pdpStatus.setDueDate(disbursementVoucherDocument.getDisbursementVoucherDueDate());
+        pdpStatus.setDvStatus(disbursementVoucherDocument.getDocumentHeader().getWorkflowDocument().getStatus().getLabel());
         return pdpStatus;
     }
-
+ 
+    @Override
+    public Set<String> cancelPDPPayments(RecurringDisbursementVoucherDocument recurringDisbursementVoucherDocument, String cancelMessage) {
+        Collection<PaymentDetail> paymentDetails = findPaymentDetailsFromRecurringDisbursementVoucher(recurringDisbursementVoucherDocument);
+        Set<String> canceledPaymentGroups = new HashSet<String>();
+        for (PaymentDetail detail : paymentDetails) {
+            String paymentDetailStatus = detail.getPaymentGroup().getPaymentStatusCode();
+            boolean doCancel = paymentDetailStatus != null && 
+                    isPaymentCancelable(GlobalVariables.getUserSession().getPerson(), paymentDetailStatus);
+            if (doCancel) {
+                LOG.debug("cancelPDPPayments: About to Cancel " + detail.getId() + " for a reason of " + cancelMessage);
+                canceledPaymentGroups.add(detail.getPaymentGroupId().toString());
+                getRecurringDisbursementVoucherPaymentMaintenanceService().cancelPendingPayment(detail.getPaymentGroupId().intValue(), 
+                        detail.getId().intValue(), cancelMessage, GlobalVariables.getUserSession().getPerson());
+            } else {
+                LOG.debug("cancelPDPPayments: Not going to cancel "+ detail.getId());
+            }
+        }
+        noteChangeOnRecurringDV(recurringDisbursementVoucherDocument, "The following payments were canceled: ", canceledPaymentGroups);
```

Comment:
Should you add this note only if there were some payments cancelled? I could be confusing to the user if a note is added with an empty list.